src/main/java/org/bih/eos/converter/cdm_field/date/DVToDateConverter.java

```diff
@@ -27,63 +27,76 @@ public class DVToDateConverter {
 
     private static final Logger LOG = LoggerFactory.getLogger(DVToDateConverter.class);
 
-    public Optional<Date> convert(Object objectAtPath) throws UnprocessableEntityException {
-        Class objectClassAtPath = objectAtPath.getClass();
-        if (objectClassAtPath.equals(PointEvent.class)) {
-            PointEvent pointEvent = (PointEvent) objectAtPath;
-            return parseDvDateTime(pointEvent.getTime().getValue());
-        } else if (objectClassAtPath.equals(IntervalEvent.class)) {
-            return getIntervalEventStart((IntervalEvent) objectAtPath);
-        } else if (objectClassAtPath.equals(Observation.class)) {
-            Observation observation = (Observation) objectAtPath;
-            return parseDvDateTime(observation.getData().getOrigin().getValue());
-        } else if (objectClassAtPath.equals(Element.class) && ((Element) objectAtPath).getValue().getClass().equals(DvDateTime.class)) {
-            DvDateTime dvDateTime = (DvDateTime) ((Element) objectAtPath).getValue();
-            return parseDvDateTime(dvDateTime.getValue());
-        } else if (objectClassAtPath.equals(Element.class) && ((Element) objectAtPath).getValue().getClass().equals(DvDate.class)) {
-            DvDate dvDate = (DvDate) ((Element) objectAtPath).getValue();
-            return parseDvDate(dvDate.getValue());
-        } else if (objectClassAtPath.equals(Action.class)) {
-            Action action = (Action) objectAtPath;
-            return parseDvDateTime(action.getTime().getValue());
-        } else if (objectClassAtPath.equals(EventContext.class)) {
-            EventContext eventContext = (EventContext) objectAtPath;
-            return parseDvDateTime(eventContext.getStartTime().getValue());
-        }
-        LOG.warn("A Date type was found that was currently not supported or can not be mapped ! Mapping will be processed if optional otherwise the conversion is skipped and the next one processed");
-        return Optional.empty();
+    public Optional<Date> convert(Object objectAtPath)  {
+    	
+        try {
+			Class objectClassAtPath = objectAtPath.getClass();
+			if (objectClassAtPath.equals(PointEvent.class)) {
+			    PointEvent pointEvent = (PointEvent) objectAtPath;
+			    return parseDvDateTime(pointEvent.getTime().getValue());
+			} else if (objectClassAtPath.equals(IntervalEvent.class)) {
+			    return getIntervalEventStart((IntervalEvent) objectAtPath);
+			} else if (objectClassAtPath.equals(Observation.class)) {
+			    Observation observation = (Observation) objectAtPath;
+			    return parseDvDateTime(observation.getData().getOrigin().getValue());
+			} else if (objectClassAtPath.equals(Element.class) && ((Element) objectAtPath).getValue().getClass().equals(DvDateTime.class)) {
```

Comment:
we could make a if on element and pack the other element ifs inside of it.
Would improve readability.